internal/services/solution/api_solution.go

```diff
@@ -28,6 +28,19 @@ type Client struct {
 	Api *api.Client
 }
 
+// buildApiUrl builds a URL for API endpoints with a given host, path, and query parameters.
+func buildApiUrl(host, path string, query url.Values) string {
```

Comment:
don't duplicate those functions, just more it to helpers and reuse it in all services